src/core/storage.rs

```diff
@@ -131,7 +135,7 @@ pub struct RecordsStorage<K: Hash + Clone> {
 	/// The last block number we've seen. Used to index the storage of all entries.
 	last_block: Option<BlockNumber>,
 	/// Elements with expire dates.
-	ephemeral_records: BTreeMap<BlockNumber, HashMap<K, Arc<StorageEntry>>>,
+	ephemeral_records: BTreeMap<BlockNumber, HashMap<K, Weak<StorageEntry>>>,
```

Comment:
Can you document the usage of `Weak` here please. I am not sure it is really needed.

Reply:
Please see the update.